File: 프로그래머스/lv2/12973. 짝지어 제거하기/짝지어 제거하기.java
import java.util.*;
class Solution
{
    public int solution(String s)
    {
        int answer = 0;
        
        char input[] = s.toCharArray();
        Stack<Character> stack = new Stack<>();
        
        for(int i = 0;i < input.length;i++) {
            if(stack.isEmpty()) stack.push(input[i]);
            else {
                if(stack.peek() == input[i]) {
                    stack.pop();
                } else stack.push(input[i]);
            }
        }
        
        answer = stack.isEmpty() == true ? 1 : 0;
        
        return answer;
    }
}

File: 프로그래머스/2/12973. 짝지어 제거하기/짝지어 제거하기.js
function solution(s)
{
    var answer = -1;

    const stack = [];
    
    for(let i = 0;i < s.length;i++) {
        if(stack.length === 0) stack.push(s[i]);
        else {
            if(stack[stack.length - 1] === s[i]) stack.pop();
            else stack.push(s[i]);
        }
    }
    
    if(stack.length === 0) answer = 1;
    else answer = 0;
 
    return answer;
}